mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutor.java

```diff
@@ -451,6 +454,7 @@ <T> CompletableFuture<T> callInMainThread(Callable<CompletableFuture<T>> tSuppli
     public CompletableFuture<Ack> submitTask(ExecuteStageRequest request) {
 
         log.info("Received request {} for execution", request);
+        this.currentRequest = request;
```

Comment:
nit: I would also move this to prepareTask() since that's where the rest of the business logic is handled with respect to the state changes.